cicecore/cicedyn/dynamics/ice_dyn_vp.F90

```diff
@@ -2552,8 +2552,14 @@ function global_dot_product (nx_block  , ny_block , &
       enddo
       !$OMP END PARALLEL DO
 
-      ! Use local summation result unless bfbflag is active
-      if (bfbflag == 'off') then
+      ! Use faster local summation result for several bfbflag settings.
+      ! The local implementation sums over each block, sums over local
+      ! blocks, and calls global_sum on a scalar and should be just as accurate as
+      ! bfbflag = 'off', 'lsum8', and 'lsum4' without the extra copies and overhead
+      ! in the more general array global_sum.  But use the array global_sum
+      ! if bfbflag is more strict or for tripole grids (requires special masking)
+      if (ns_boundary_type /= 'tripole' .and. ns_boundary_type /= 'tripoleT' .and. &
```

Comment:
Just added this check on tripole grids.  Don't want to use the local sum with tripole grids because special masking is required.  Reran full test suite on cheyenne with gnu compiler and should be good.